pkg/alertmanager/operator.go

```diff
@@ -162,6 +156,45 @@ func (c *Operator) RegisterMetrics(r prometheus.Registerer) {
 	)
 }
 
+// waitForCacheSync waits for the informers' caches to be synced.
+func (c *Operator) waitForCacheSync(stopc <-chan struct{}) error {
+	ok := true
+	informers := map[cache.SharedInformer]bool{
```

Comment:
`cache.SharedInformer` is an interface. We are not in control over the implementing underlying type. If the underlying type uses an incomparable type (i.e. a composited map or slice) this will result in a runtime panic.

I am new to informers, but can we construct a natural key (struct) with values we have under our control?

Reply:
OK sounds good

Reply:
I think we can actually describe each informer with a string, so this shouldn't be much of an issue